src/cast_core/src/LoggingActor.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cast_core/actors/LoggingActor.hpp>
+
+#include <boost/log/trivial.hpp>
+
+#include <gennylib/Cast.hpp>
+#include <gennylib/context.hpp>
+
+namespace genny::actor {
+
+//
+// Note this Actor only has a manually-run test-case (LoggingActor_test.cpp).
+// Be careful when making changes.
+//
+
+struct LoggingActor::PhaseConfig {
+    TimeSpec logEvery;
+    metrics::clock::time_point started;
+    unsigned iteration = 0;
+
+    explicit PhaseConfig(PhaseContext& phaseContext)
+        : logEvery{phaseContext["LogEvery"].to<TimeSpec>()}, started{metrics::clock::now()} {
+        if (phaseContext["Blocking"].to<std::string>() != "None") {
+            BOOST_THROW_EXCEPTION(
+                InvalidConfigurationException("LoggingActor must have Blocking:None"));
+        }
+    }
+    void report() {
+        // Avoid calling now() on most iterations
+        if (++iteration % 10000 != 0) {
```

Comment:
It should be possible to add a `__builtin_expect` compiler hint here if you want to be extra-fancy, but probs not worth it.

Reply:
Ya so far we've stayed pretty much entirely on non-vendor-specific C++17 so I'm hesitant to venture into extensions for something as simple as this.